### ISSUE TITLE
Heading should support `aria-level` prop override

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#5077]
### Provide a general summary of the feature here

Heading component needs to support `aria-level` override to set the hierarchical heading level in the accessibility tree regardless of the rendered element.

### 🤔 Expected Behavior?

A user should be able to explicitly set the `aria-level` for a `<Heading>` instance, like:
```jsx
  <Heading level={3} aria-level={2}>Heading content</Heading>
```

This is necessary for use cases where the Heading `level` gets adjusted for responsive design reasons, but the `aria-level` should remain unchanged.

### 😯 Current Behavior

There is currently no way to override the `aria-level` of a Heading component:

```jsx
  <Heading level={3} aria-level={2}>Heading content</Heading>
```

will render as 

```html
  <h3 aria-level="2">Heading content</h3>
```

regardless of its intended heading level within the hierarchy of the web page.

### 🔦 Context

Noticed a use case in the `@quarry/inventory` `Title` component that the `level` for a Heading is changing depending on responsive layout.

```jsx
export function Title(props: TitleProps): ReactElement {
  const {children} = props;
  const { setTitleId } = useInventoryContext();
  const titleId = useId();
  useEffect(() => setTitleId(titleId), []);
  const {matchedBreakpoints} = useBreakpoint();
  const level = getResponsiveProp({base: 3, L: 2}, matchedBreakpoints) as 2 | 3;
  return (
    <Heading gridArea="title" id={titleId} level={level}>
      {children}
    </Heading>
  );
}
```

### 🧢 Your Company/Team

Adobe/Accessibility

**Comment by snowystinger** [comment]
Just to clarify, you're saying it should render like this, given your example?
```
<h3 aria-level="2">Heading content</h3>
```

**Comment by devongovett** [comment]
I think the `level` prop was intended to support changing the HTML element _without_ changing the styling. Currently, `Heading` does not have any styling associated with it by itself - that would come from e.g. dialog. So I think this should already be the case?

**Comment by majornista** [comment]
> I think the `level` prop was intended to support changing the HTML element _without_ changing the styling. Currently, `Heading` does not have any styling associated with it by itself - that would come from e.g. dialog. So I think this should already be the case?

That's not how it seems to be used in practice, as the example I reference above demonstrates.

**Comment by devongovett** [comment]
Yeah if it isn't defining any styling, then it'll just get the default font size for the heading element from the browser I think.

**Comment by majornista** [comment]
Closing as working as intended. See frontend/quarry/issues/5376 for suggested change in Quarry.
